server/src/main/java/org/elasticsearch/cluster/metadata/ComposableIndexTemplate.java

```diff
@@ -250,11 +250,11 @@ public String toString() {
 
         private static final ConstructingObjectParser<DataStreamTemplate, Void> PARSER = new ConstructingObjectParser<>(
             "data_stream_template",
-            args -> new DataStreamTemplate((String) args[0])
+            args -> new DataStreamTemplate(args[0] != null ? (String) args[0] : FIXED_TIMESTAMP_FIELD)
         );
 
         static {
-            PARSER.declareString(ConstructingObjectParser.constructorArg(), DataStream.TIMESTAMP_FIELD_FIELD);
+            PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), DataStream.TIMESTAMP_FIELD_FIELD);
```

Comment:
maybe remove this field from the xcontent format? So that the only valid format is an empty json object?

Reply:
Yes, good point. i'll drop the field completely in the POJO but keep the `data_stream` empty JSON object